src/workbookManager.ts

```diff
@@ -85,21 +85,39 @@ export class WorkbookManager {
     }
 
     private async updateSingleQueryAttributes(zip: JSZip, queryName: string, refreshOnOpen: boolean) {
-        const refreshOnLoadValue = refreshOnOpen ? "1" : "0";
-        const connectionId = await this.editConnections(zip, queryName, refreshOnLoadValue);
-        const sharedStringId = (await this.editSharedStrings(zip, queryName)).toString();
-        await this.editWorksheet(zip, sharedStringId);
-        await this.editPivotTable(zip, queryName, refreshOnLoadValue, connectionId);  
-    }
-
-    private async editConnections(zip: JSZip, queryName: string, refreshOnLoadValue: string) {
+        //Update connections
         const connectionsXmlString = await zip.file(connectionsXmlPath)?.async("text");
         if (connectionsXmlString === undefined) {
             throw new Error("Connections were not found in template");
+        }  
+        
+        const {connectionId, connectionString} = await this.updateConnections(connectionsXmlString, queryName, refreshOnOpen);
+        zip.file(connectionsXmlPath, connectionString);
+        
+        //Update sharedStrings
+        const sharedStringsXmlString = await zip.file(sharedStringsXmlPath)?.async("text");
+        if (sharedStringsXmlString === undefined) {
+            throw new Error("SharedStrings were not found in template");
+        }
+        const {sharedStringIndex, newSharedStrings} = await this.updateSharedStrings(sharedStringsXmlString, queryName);
+        zip.file(sharedStringsXmlPath, newSharedStrings);
+        
+        //Update sheet
+        const sheetsXmlString = await zip.file(sheetsXmlPath)?.async("text");
+        if (sheetsXmlString === undefined) {
+            throw new Error("Sheets were not found in template");
         }
+        const worksheetString = await this.updateWorksheet(sheetsXmlString, sharedStringIndex.toString());
+        zip.file(sheetsXmlPath, worksheetString);
+        
+        //Update tables
+        await this.updatePivotTables(zip, queryName, refreshOnOpen, connectionId);  
```

Comment:
I assume we can move to change updatePivotTables to a similar testable pattern in a separate PR